### DISCUSSION TITLE
Symfony plugin for vs code

### DISCUSSION THREAD

**Discussion by armanhadifatah** [symfony/symfony#46793] [Polls]
Hello dear 
Why there is no symfony plugin for vs code as a developer now i and  many developer use vs code for all kind of developments.
 
I think a good symfony plugin for vs code will help a lot of developers  and most importantly new symfony developers

**Comment by lyrixx** [comment]
What would do the plugin?

---

BTW, did you see there are already many extensions?

![image](https://user-images.githubusercontent.com/408368/176214981-1d586480-3b10-4cfe-a72a-68782289b500.png)

**Reply by armanhadifatah** [reply]
All of them are old and not maintained

**Comment by armanhadifatah** [comment]
for helping with code like the symfony plugin for phpstorm

**Reply by DylanKas** [reply]
There are quite a few symfony specific plugins, as well as PHP IntelliSense to fix a lot of thing as well.
However, I tried both VS Code and PHPStorm and I feel like PHPStorm is by far better to use.
It's probably possible to get something close to PHPStorm with VS code by customizing A LOT your ide, but PHPStorm is just great out of the box.

**Comment by ghost** [comment]
First of all, VS Code is not an IDE, PHPStorm is.VS Code is a code editor, it's like Sublime on stereoids. It's also lightweight whereas IDE'a are heavyweight and consume a lot of system memory. VS Code was not created for PHP but rather for HTML, JS, Go, Python - I mean lightweight languages. PHP is a combine. 

You can read how VS Code describes itself here:
> Visual Studio Code is a streamlined code editor with support for development operations like debugging, task running, and version control. It aims to provide just the tools a developer needs for a quick code-build-debug cycle and leaves more complex workflows to fuller featured IDEs, such as [Visual Studio IDE](https://visualstudio.microsoft.com/).

P.S. IIRC @fabpot uses VS Code, so maybe he can give you some guides how to improve your work on PHP project with VS Code.

**Reply by armanhadifatah** [reply]
Is dart a lightweight language ? that doesn't make any sense to me .
what flutter team did with vs code plugin is great and you do not need any
other editor for flutter framework , i know it takes time for symfony team
to make a plugin like this , but it will help them to get developer to use
the framework more , there are many thing that vs code do not
understand about symfony but if there is a good plugin for it ti will help
alot

**Reply by ghost** [reply]
It's all about the community. PHP community mostly uses well known and battle tested PHPStorm that was created especially for PHP and has very good Symfony plugin so don't expect any sophisticated plugins for VS Code. 

> i know it takes time for symfony team to make a plugin like this
Symfony team does not provide features like this.

**Reply by IArnaut2** [reply]
That VS Code is created for HTML, JS, Python is irrelevant. If it can be used to make PHP projects, then it should have Symfony plugins.

**Comment by armanhadifatah** [comment]
Is dart a lightweight language ? that doesn't make any sense to me .
what flutter team did with vs code plugin is great and you do not need any
other editor for flutter framework , i know it takes time for symfony team
to make a plugin like this , but it will help them to get developer to use
the framework more , there are many thing that vs code do not
understand about symfony but if there is a good plugin for it ti will help
alot

**Comment by lyrixx** [comment]
I'm not sure to understand this whole discussion. I'm using code for years and I can tell you it works very well with PHP. Of course it can be better.

I don't know and I don't care if PHP was in the initial plan of vs code. That does not matter. The support is really good! And don't forget that PHP storm comes from intellij that was build for Java. So this argument is not relevant. 

And about if it's an ide or not... You have all the same features (code completion, step by step debugger, terminal, git integration, remote control, snippet, macro, code sharing, etc) as others ides. I don't know what miss you. 

**So let please focus on the initial issue**, and don't create a debate between code and phpstorm. Both are good and let's focus only on code here.  

Do we need a Symfony plugin? Probably. It would help everyone. New comers and experienced developers.

But there are already plugins. Someone told the are old. So What the issue? PHP is old and we all use it!
Are they unmaintained ? I Don't know. But if it's the case, I think it's better to help the maintainer than to create a new one.

I don't know if someone in the core team want to maintain such plugin, we have already too much work. 

If someone here want to help, please do. It Would be awesome 😎

**Reply by faizanakram99** [reply]
@lyrixx Laravel just launched a plugin for vscode the last week at Laracon, it seems as good as PhpStorm's symfony plugin.

PhpStorm while a good IDE is resource intensive and it is buggy (especially when used with WSL ... remote gateway or wslg), VS code has top notch remote development experience, an official symfony plugin would really make a huge difference. It will also make it more approachable for newcomers imo.

**Reply by IArnaut2** [reply]
Perhaps there needs to be an OFFICIAL extension.

**Comment by fullstackdev610** [comment]
There are many VS code plugins for Symfony.

**Comment by armanhadifatah** [comment]
Why most say there are plugin for symfony , that is
nonsense all are old and no use for new symfony versions , what I mean
offcial symfony plugin not alot of plugin which they can't offer good value
for symfony , for example flutter plugin for vs code is realy amazing and
its by flutter team we need something like that.

**Reply by lyrixx** [reply]
Feel free to modernize theses plugins 👍🏼

**Reply by IArnaut2** [reply]
> Feel free to modernize theses plugins 👍🏼

How about you?

**Comment by andrescevp** [comment]
I am going to just drop this here: https://marketplace.visualstudio.com/items?itemName=AndresCevallos.vscode-symfonykd-devkit

What is Symfonykd -> Symfony kit For Developers? 
Is my personal approach to solve development experience (my experince)

Now... I am publishing all under previews/pre-relase version, the stable version have exactly that, the first version - not really useful, looking for a closed group of users to help to refine the extension.

The future? 
I have a clear roadmap... Let's see if the extension get's popular to invest time un the real roadmap, not only the things I need/use. Ultimately I would love keep the development of this going forward... but let's see.

Hope you like my small contribution to the Symfony ecosystem!

**Reply by lyrixx** [reply]
It looks like your repository is closed source (I got a 404)

**Reply by andrescevp** [reply]
the repository is closed because is not open source, I am goingt to setup a public website and use that as repository, the real code of the extension is not going to be published

**Reply by andrescevp** [reply]
@lyrixx btw, happy to provide temporal access to core symfony team so you guys can evaluate the extension and provide direction :)
